shared/utils/graphql.ts

```diff
@@ -0,0 +1,280 @@
+import type {
+  AssertionGraphEntity,
+  ChainId,
+  ChainName,
+  OOV2GraphEntity,
+  OracleType,
+  ParsedOOV1GraphEntity,
+  ParsedOOV2GraphEntity,
+  PriceRequestGraphEntity,
+  RequestState,
+} from "@shared/types";
+import { BigNumber } from "ethers";
+
+export function handleGraphqlNullableStringOrBytes<
+  EntityKey extends { [key: string]: string | null }
+>(entityKey: EntityKey | null) {
+  if (entityKey === null) return null;
+
+  const [property, stringOrBytes] = Object.entries(entityKey)[0];
+
+  if (!(typeof stringOrBytes === "string" || stringOrBytes === null)) {
+    console.error(
+      `Wrong type in \`handleGraphqlNullableStringOrBytes\`: Expected string or null for ${property}, got ${typeof stringOrBytes} instead`
+    );
+  }
+  if (stringOrBytes === null) return null;
+
+  return stringOrBytes;
+}
+
+export function handleGraphqlNullableBigInt<
+  EntityKey extends { [key: string]: string | null }
+>(entityKey: EntityKey | null) {
+  if (entityKey === null) return null;
+
+  const [property, bigInt] = Object.entries(entityKey)[0];
+
+  if (!(typeof bigInt === "string" || bigInt === null)) {
+    console.error(
```

Comment:
why cant we just throw these errors?

Reply:
two reasons:

throwing the error here fails silently unless you catch it elsewhere (easy to forget)
just throwing the error means you can only see the value and not the key, so you'll just get an `undefined` for example but you won't know which key was undefined. this way you'll know which key has a wrong value